src/resources/index.ts

```diff
@@ -28,38 +34,60 @@ export default async function(app: Express, db: Db) {
   });
   
   /* Resource Life Cycle Routes */
-  app.post('/api/v1/resources', async (req, res) => {
-      const { insertedId } = await db.collection('resources').
-        insertOne(req.body);
-      res.status(201);
-      res.send(JSON.stringify({ status: 201, id: insertedId }));
-  });
-  
-  app.put('/api/v1/resources/:id', async (req, res) => {
-      const id = req.params.id;
 
-      // create an update doc for MongoDB
-      const updateDoc = {
-        $set: {...req.body}
-      }
+   /**
+   * Dummy route because GUI wants a content upload URL, even when we're just copying a YouTube link
+   * TO DO: remove the upload content requirement from the GUI, so that YouTube (or other web) links no longer have to post to an dummy file upload link
+   */
+    app.post('/api/v1/resources/:id/content/:token', authorizeRequest, async (req, res) => {
+      return res.status(200).send({
+        status: 200,
+        id: req.params.id
+      })
+  });
 
-      const response = await db.collection('resources').
-        updateOne({"_id": new ObjectID(id)}, updateDoc);
+  /**
+   * POST takes a resource object (see schema), and returns a object with status, id, and a contentUploadUrl string used by the GUI to fake uploading content (see post '/api/v1/resources/:id/content/:token')
+   */
 
-      res.status(201);
-      res.send(JSON.stringify({ matches: response.matchedCount, mondified: response.modifiedCount}));
+  app.post('/api/v1/resources', authorizeRequest, async (req, res) => {
+    Resource.create(req.body, function (err, resource) {
+      if (err) { res.status(500).send(JSON.stringify({status: 500, err})); }
+      else {
+        res.status(200);
+        res.send(JSON.stringify({status: 200, resource: {id: resource._id}, contentUploadUrl: `/api/v1/resources/${resource._id}/content/dummytoken-${resource._id}`}));
```

Comment:
`contentUploadUrl` needs to contain the whole URL. I think that's why you were getting scheme errors. It would be trivial to change ayamel to not require this bit if we decide to move in that direction.

Reply:
I would actually lean the direction of updating the GUI to use the base resource URL in the config. The GUI already uses a base URL for other calls, not sure why it wouldn't for this one.